lib/compilers/rga.ts

```diff
@@ -0,0 +1,205 @@
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import path from 'path';
+
+import {readdir, rename, writeFile} from 'fs-extra';
+
+import {ExecutionOptions} from '../../types/compilation/compilation.interfaces';
+import {ParseFilters} from '../../types/features/filters.interfaces';
+import {BaseCompiler} from '../base-compiler';
+import * as exec from '../exec';
+import {logger} from '../logger';
+
+interface ASICSelection {
+    asic?: string;
+    error?: string;
+    printASICs?: boolean;
+}
+
+// RGA := Radeon GPU Analyzer
+export class RGACompiler extends BaseCompiler {
+    private dxcPath: string;
+
+    static get key() {
+        return 'rga';
+    }
+
+    constructor(info: any, env: any) {
+        super(info, env);
+
+        this.compiler.supportsIntel = false;
+        this.dxcPath = this.compilerProps(`compiler.${this.compiler.id}.dxcPath`);
+        logger.debug(`RGA compiler ${this.compiler.id} configured to use DXC at ${this.dxcPath}`);
+    }
+
+    override optionsForFilter(filters: ParseFilters, outputFilename: any, userOptions?: any): any[] {
+        return [outputFilename];
+    }
+
+    extractASIC(dxcArgs: string[]): ASICSelection {
+        // Scan dxc args for an `--asic` argument that should be stripped and passed later to RGA
+        // Default to RDNA2
+        let asic = 'gfx1030';
+        let printASICs = true;
+        for (let i = 0; i !== dxcArgs.length; ++i) {
+            const arg = dxcArgs[i];
+            if (arg === '--asic') {
+                // NOTE: the last arguments are the input source file and -spirv, so check
+                // if --asic immediately precedes that
+                if (i === dxcArgs.length - 3) {
+                    return {
+                        error: '--asic flag supplied without subsequent ASIC!',
+                    };
+                }
+                asic = dxcArgs[i + 1];
+                // Do a quick sanity check to determine if a valid ASIC was supplied
+                if (!asic.startsWith('gfx')) {
+                    return {
+                        // eslint-disable-next-line max-len
```

Comment:
Could you not disable this eslint, and instead consider breaking onto two lines? It's the same number of lines but doesn't normalise the "turn the warnings off" as a default :)